common/network-common/src/main/java/org/apache/spark/network/client/TransportResponseHandler.java

```diff
@@ -188,6 +188,7 @@ public void handle(ResponseMessage message) throws Exception {
       if (listener == null) {
```

Comment:
While we're here, I wonder if it's better to extend the whole try-finally block around this else-if branch? what if outstandingRpcs.get/remove fails? highly unlikely I guess so any 'leak' is trivial.

But what about the same problem around line 168 and 172? there's no try-finally around onSuccess, which could fail?

Do all of these messages really need to be release()-ed?

Reply:
`I wonder if it's better to extend the whole try-finally block around this else-if branch? what if outstandingRpcs.get/remove fails? highly unlikely I guess so any 'leak' is trivial.` ,  Yes , the outstandingRpcs.get/remove can not fail, so we should not try-finally block around this else-if branch here, which may Improve performance.

`But what about the same problem around line 168 and 172? there's no try-finally around onSuccess, which could fail?`, i think we should  also try-finally around onSuccess around line 168 and 172, do you think so ?

`Do all of these messages really need to be release()-ed? meaning, should .release() happen in all branches?`,It should be released, although this scenario is not easy to happen.

Reply:
I just wonder if it would be simpler to wrap a lot more in one big try-finally and release the thing in all cases?